DM-30866: add support for updating dimension records

TallJimbo opened the pull request:
## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

timj commented on python/lsst/obs/base/defineVisits.py:
```diff
@@ -279,6 +279,18 @@ class DefineVisitsConfig(Config):
         optional=False,
         default=True,
     )
+    update_records = Field(
```
Aren't we going to want to surface these in the command-line tooling for ingest-raw and define-visit? Does `butler ingest-raw` override the config flag? Is this really a config option or is it a parameter for `run`?

timj reviewed (changes requested):
Two main points:

* The command-line versions need these abilities and do we think that these are config options versus parameters?
* We do have fairly comprehensive test coverage to ingest-raws because there are JSON index files. This means you should be able to test this by ingesting with the default index file and then ingesting again with a modified index file.

So I'll defer approving this just for the moment.

TallJimbo commented:
> The command-line versions need these abilities and do we think that these are config options versus parameters?

I don't really have a good rubric for when to use a config parameter vs. a `run` or `__init__` argument for these two tasks, or others like them that are run from the command-line and aren't `PipelineTasks` (though there are some clear-cut cases, where per-instrument overrides or not-representable-in-config objects in play).  Using configs is certainly an easier way to add and reuse options, especially if (as I thought) we automatically get command-line support there, and I think it's a lot less error-prone than editing/reading the ~5 billion files you have to modify consistently to add a single click option.  But custom click options are certainly a bit prettier on the command line.

So if you've got a clear preference on where to put these, I'm happy to go with that.

It may also be worth considering whether we _want_ these on the command-line interface; they're not for everyday use, and I would personally be inclined to use them only when I can very carefully feed them inputs in Python.  But I also get the sense that I'm kind of unusual in wanting to do all (not just most) of my scripting in Python.

> We do have fairly comprehensive test coverage to ingest-raws because there are JSON index files. This means you should be able to test this by ingesting with the default index file and then ingesting again with a modified index file.

Will do.

TallJimbo commented:
Actually, at least for `RawIngestTask`, making these flags arguments to `run` will make it a lot easier to test them, the way the tests are set up, and that's enough to swing me in their favor.  So still happy to go with your preference on where to put these, but that's what I'll do if you don't have a preference.

timj commented:
Parameters are fine with me. I'm okay with not immediately making them visible to the command-line. You could have a config item and allow parameter override of course (and just move the config read to the `run` method).

TallJimbo commented:
Ok, I think I've addressed comments: I've converted the config options to method arguments and added tests for `RawIngestTask`.  There's no support for using them via the CLI, but I think that's probably for the best and it's easier to add things later than take them away.

timj reviewed (approved):
Looks great. Thanks.
